### PR TITLE
Fix comment ID parameter for get_edit_comment_link filter.

### PR DESCRIPTION
**Pull request by felixarntz**
This is a follow-up fix that addresses the bug in https://core.trac.wordpress.org/ticket/61727#comment:10.

Trac ticket: https://core.trac.wordpress.org/ticket/61727

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by peterwilsoncc** [comment]
@felixarntz This looks good to me but are you able to do a force push of just the test, wait for the actions to complete and then push the fix?

**Comment by felixarntz** [comment]
@peterwilsoncc Done, please see e.g. https://github.com/WordPress/wordpress-develop/actions/runs/10802717139/job/29965357479?pr=7322 for showing the failing test prior to the fix being re-added.

File: src/wp-includes/link-template.php [hunk 1]
```diff
@@ -1618,14 +1618,17 @@ function get_edit_comment_link( $comment_id = 0, $context = 'display' ) {
...
+
 	/**
 	 * Filters the comment edit link.
 	 *
 	 * @since 6.7.0 The $comment_id and $context parameters are now being passed to the filter.
 	 *
 	 * @param string $location The edit link.
 	 * @param int    $comment_id Optional. Unique ID of the comment to generate an edit link.
-	 * @param int    $context    Optional. Context to include HTML entities in link. Default 'display'.
+	 * @param string $context    Optional. Context to include HTML entities in link. Default 'display'.
```
Comment (felixarntz) [inline src/wp-includes/link-template.php:1631]: This is an unrelated minor fix for an incorrect doc type that I noticed while fixing the main bug here.

File: src/wp-includes/link-template.php [hunk 2]
```diff
@@ -1618,14 +1618,17 @@ function get_edit_comment_link( $comment_id = 0, $context = 'display' ) {
...
 	 * Filters the comment edit link.
 	 *
 	 * @since 6.7.0 The $comment_id and $context parameters are now being passed to the filter.
 	 *
-	 * @param string $location The edit link.
-	 * @param int    $comment_id Optional. Unique ID of the comment to generate an edit link.
-	 * @param int    $context    Optional. Context to include HTML entities in link. Default 'display'.
+	 * @param string $location   The edit link.
+	 * @param int    $comment_id Unique ID of the comment to generate an edit link.
+	 * @param string $context    Context to include HTML entities in link. Default 'display'.
```
Comment (felixarntz) [inline src/wp-includes/link-template.php:1631]: These are unrelated minor fixes for an incorrect doc type and indentation that I noticed while fixing the main bug here.

**Review by peterwilsoncc — APPROVED** [review]
Thanks Felix, looks good to me,

**Review by mukeshpanchal27 — APPROVED** [review]

**Review by dream-encode — APPROVED** [review]

**Comment by felixarntz** [comment]
Committed in https://core.trac.wordpress.org/changeset/59012
